lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -24,6 +24,12 @@ const String transparentTextEditingClass = 'transparentTextEditing';
 
 void _emptyCallback(dynamic _) {}
 
+/// The root node that hosts all DOM required for text editing.
+///
+/// This is something similar to [html.Document]. Currently, it's a [html.ShadowRoot].
+@visibleForTesting
+html.ShadowRoot get textEditingRoot => html.document.querySelector('flt-glass-pane')!.shadowRoot!;
```

Comment:
We don't want to rely on `querySelector` for anything except debugging (e.g. `assert`). It's slow and it's a step backwards from our goal of supporting multiple Flutter views on the same page. Can we instead have the `DomRenderer` expose `glassPaneShadow`?

Reply:
Yes, let me try this. It was causing some issues earlier, but I'll try to expose the glassPaneShadowRoot as you suggested.

Reply:
Looks like it's working now. I'm not sure how I messed it up so bad (it might be messed up in CI still)

Anyway, I'll check the breaking tests and push with all the changes!